Panic when pulling from certain (closed) mensas

```
panic: invalid h3 tag

goroutine 39 [running]:
github.com/lukasmoellerch/mensa-cli/internal/uzh.fetchMenuUzh.func1(0x81f9cd, 0xc0004226c0)
	/home/aaron/Projects/mensa-cli/internal/uzh/provider.go:277 +0x94b
github.com/PuerkitoBio/goquery.(*Selection).EachWithBreak(0xc000422450, 0xc000161e10)
	/home/aaron/go/pkg/mod/github.com/!puerkito!bio/goquery@v1.8.0/iteration.go:21 +0x11f
github.com/lukasmoellerch/mensa-cli/internal/uzh.fetchMenuUzh({0x99781e, 0x5}, {0xc00002cfc0, 0xf}, {0xc00002cfe0, 0xd}, {0x996414, 0x0}, {0x99c8fa, 0xa})
	/home/aaron/Projects/mensa-cli/internal/uzh/provider.go:232 +0x53a
github.com/lukasmoellerch/mensa-cli/internal/uzh.(*Provider).FetchMenus.func1()
	/home/aaron/Projects/mensa-cli/internal/uzh/provider.go:79 +0x12b
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/aaron/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x67
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/aaron/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x92
```

caused by [this](https://www.mensa.uzh.ch//en/menueplaene/lichthof-rondell/donnerstag.html) because there's no pipe in the h3 element. Fix by not panicking, but just warning (or returning false) when that happens.

--- a/cmd/group-add.go
+++ b/cmd/group-add.go
@@ -32,7 +32,7 @@ This command will open an editor to allow you to add the group's caanteens.`,
 
 		store, err := base.NewStore(storageDirectory)
 		if err != nil {
-			panic(err)
+			return err
 		}
 		if err := syncStore(ctx, store); err != nil {
 			return err

--- a/cmd/group-update.go
+++ b/cmd/group-update.go
@@ -24,7 +24,7 @@ var groupUpdateCmd = &cobra.Command{
 
 		store, err := base.NewStore(storageDirectory)
 		if err != nil {
-			panic(err)
+			return err
 		}
 		if err := syncStore(ctx, store); err != nil {
 			return err

--- a/cmd/meals.go
+++ b/cmd/meals.go
@@ -19,6 +19,7 @@ var mealsFilterFlag string
 var mealsGroupFlag string
 var mealsDinnerFlag bool
 var mealsLunchFlag bool
+var mealsAllFlag bool
 var mealsTomorrowFlag bool
 
 // mealsCmd represents the get command
@@ -40,6 +41,13 @@ If the filter is empty, all facilities are fetched. The filter is matched agains
 		if mealsGroupFlag != "" && mealsFilterFlag != "" {
 			return fmt.Errorf("cannot use both --group and --filter")
 		}
+		if mealsGroupFlag != "" && mealsAllFlag {
+			return fmt.Errorf("cannot use both --group and --all")
+		}
+		if mealsFilterFlag != "" && mealsAllFlag {
+			return fmt.Errorf("cannot use both --filter and --all")
+		}
+
 		if mealsDinnerFlag && mealsDaytimeFlag != "" {
 			return fmt.Errorf("cannot use both --dinner and --daytime")
 		}
@@ -88,7 +96,7 @@ If the filter is empty, all facilities are fetched. The filter is matched agains
 		}
 
 		filtered := []*storage.CanteenData{}
-		if mealsFilterFlag != "" {
+		if mealsFilterFlag != "" || mealsAllFlag {
 			filtered, err = store.Filter(ctx, langFlag, mealsFilterFlag)
 			if err != nil {
 				return err
@@ -205,12 +213,16 @@ func renderResult(results map[string]base.CanteenMenu, canteenLabels map[string]
 		if !ok {
 			label = result.Canteen
 		}
+
 		s += fmt.Sprintf("\n%s\n\n", canteenNameStyle.Render(label))
+
 		for i, meal := range result.Meals {
 			if i != 0 {
 				s += "\n"
 			}
-			s += fmt.Sprintf("%s\n", headingStyle.Render(meal.Label))
+			if meal.Label != "" {
+				s += fmt.Sprintf("%s\n", headingStyle.Render(meal.Label))
+			}
 			for _, line := range meal.Description {
 				s += fmt.Sprintf("%s\n", descriptionStyle.Render(strings.TrimSpace(line)))
 			}
@@ -237,5 +249,6 @@ func init() {
 	mealsCmd.Flags().StringVarP(&mealsGroupFlag, "group", "g", "", "Group to fetch canteens from")
 	mealsCmd.Flags().BoolVarP(&mealsDinnerFlag, "dinner", "i", false, "Fetch dinner meals")
 	mealsCmd.Flags().BoolVarP(&mealsLunchFlag, "lunch", "l", false, "Fetch lunch meals")
+	mealsCmd.Flags().BoolVarP(&mealsAllFlag, "all", "a", false, "Fetch meals for all canteens")
 	mealsCmd.Flags().BoolVarP(&mealsTomorrowFlag, "tomorrow", "m", false, "Fetch tomorrow's meals")
 }

--- a/internal/uzh/provider.go
+++ b/internal/uzh/provider.go
@@ -74,11 +74,11 @@ func (p *Provider) FetchMenus(ctx context.Context, caanteens []base.CanteenRef, 
 					return nil
 				}
 			} else {
-				return fmt.Errorf("invalid daytime: %s", daytime)
+				return nil
 			}
 			menu, err := fetchMenuUzh(ctx, ref.ID, slug, lang, strWeekday)
 			if err != nil {
-				return err
+				return nil
 			}
 			menus[i] = menu
 			return nil

--- a/internal/uzh/provider.go
+++ b/internal/uzh/provider.go
@@ -96,11 +96,11 @@ func (p *Provider) FetchCanteens(ctx context.Context, lang string) ([]base.Cante
 
 	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 	res, err := http.DefaultClient.Do(req)
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 	defer res.Body.Close()
 	if res.StatusCode != 200 {
@@ -109,7 +109,7 @@ func (p *Provider) FetchCanteens(ctx context.Context, lang string) ([]base.Cante
 
 	doc, err := goquery.NewDocumentFromReader(res.Body)
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 
 	list := doc.Find("ul[role=navigation] > li > ul > li > a")
@@ -197,11 +197,11 @@ func fetchMenuUzh(ctx context.Context, id string, slug string, lang string, week
 	url := fmt.Sprintf("http://www.mensa.uzh.ch/%s/menueplaene/%s/%s.html", lang, slug, weekday)
 	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		panic(err)
+		return base.CanteenMenu{}, err
 	}
 	res, err := http.DefaultClient.Do(req)
 	if err != nil {
-		panic(err)
+		return base.CanteenMenu{}, err
 	}
 	defer res.Body.Close()
 	if res.StatusCode != 200 {
@@ -210,12 +210,12 @@ func fetchMenuUzh(ctx context.Context, id string, slug string, lang string, week
 
 	doc, err := goquery.NewDocumentFromReader(res.Body)
 	if err != nil {
-		panic(err)
+		return base.CanteenMenu{}, err
 	}
 
 	divs := doc.Find(".newslist-description > div")
 	if divs.Length() != 1 {
-		panic("no div")
+		return base.CanteenMenu{}, fmt.Errorf("invalid number of divs: %d", divs.Length())
 	}
 
 	meals := []base.Meal{}
@@ -266,7 +266,8 @@ func fetchMenuUzh(ctx context.Context, id string, slug string, lang string, week
 			// Split by | - separating the name and the price
 			parts := strings.Split(s.Text(), " | ")
 			if len(parts) != 2 {
-				panic("invalid h3 tag")
+				loopErr = fmt.Errorf("invalid label: %s", s.Text())
+				return false
 			}
 			label = parts[0]
 			price := parts[1]

